k4FWCore/include/k4FWCore/DataHandle.h

```diff
@@ -90,9 +90,9 @@ template <typename T>
 DataHandle<T>::DataHandle(const std::string& descriptor, Gaudi::DataHandle::Mode a, IDataHandleHolder* fatherAlg)
     : DataObjectHandle<DataWrapper<T>>(descriptor, a, fatherAlg), m_eds("EventDataSvc", "DataHandle") {
   if (a == Gaudi::DataHandle::Writer) {
-    StatusCode sc            = m_eds.retrieve();
-    m_dataPtr                = nullptr;
-    auto* podio_data_service = dynamic_cast<PodioDataSvc*>(m_eds.get());
+    StatusCode sc [[maybe_unused]] = m_eds.retrieve();
```

Comment:
a) Why would we be ignoring this status code?

b) Doesn't this exist?

```suggestion
   m_eds.retrieve().ignore();
```

Reply:
I think that's because you get an error when trying to retrieve something that doesn't exist. 
I changed it to use `ignore()`

Reply:
The nullptr check was introduced there to be able to use the LegacyDataSvc in #117, that can be removed now. I'll add that to this PR since it's some sort of cleanup

Reply:
The nullptr check seems to be there for when another data service could be used, maybe from when there was the legacy one. I think probably the nullptr check and the check below can be removed

Reply:
But we need some way to report that something wasn't found or is this handled somewhere else?

Reply:
If something is not found in the store then that's handled by https://github.com/key4hep/k4FWCore/main/no-datasvc/k4FWCore/include/k4FWCore/DataHandle.h#L116 when using our DataHandles, if using the ones from Gaudi then it's handled by Gaudi

Reply:
Link gives "Not Found"

Reply:
Now it's correct, I got it from emacs from the branch I was using and then I have to edit the link and put `main` in the right place if I don't want to link to another branch and I put it in the wrong place